tests/core/policies/test_rule_policy.py

```diff
@@ -1928,3 +1932,277 @@ def test_predict_nothing_if_fallback_disabled():
+            PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+        },
+    ]
+
+
+def test_hide_rule_turn_with_slots():
+    some_action = "some_action"
+    some_other_action = "some_other_action"
+    some_intent = "some_intent"
+    some_other_intent = "some_other_intent"
+    some_slot = "some_slot"
+    some_slot_value = "value1"
+    some_other_slot = "some_other_slot"
+    some_other_slot_value = "value2"
+    domain = Domain.from_yaml(
+        f"""
+    intents:
+    - {some_intent}
+    - {some_other_intent}
+    actions:
+    - {some_action}
+    - {some_other_action}
+    slots:
+      {some_slot}:
+        type: text
+      {some_other_slot}:
+        type: text
+        """
+    )
+
+    simple_rule = TrackerWithCachedStates.from_events(
+        "simple rule with an action that sets 1 slot",
+        domain=domain,
+        slots=domain.slots,
+        evts=[
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": some_intent}),
+            ActionExecuted(some_action),
+            SlotSet(some_slot, some_slot_value),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+        is_rule_tracker=True,
+    )
+    simple_rule_with_slot_set = TrackerWithCachedStates.from_events(
+        "simple rule with an additional slot set before it starts",
+        domain=domain,
+        slots=domain.slots,
+        evts=[
+            SlotSet(some_other_slot, some_other_slot_value),
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": some_intent}),
+            ActionExecuted(some_action),
+            SlotSet(some_slot, some_slot_value),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+        is_rule_tracker=True,
+    )
+    simple_story_with_other_slot_set = TrackerWithCachedStates.from_events(
+        "simple rule with an additional slot set before it starts",
+        domain=domain,
+        slots=domain.slots,
+        evts=[
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": some_other_action}),
+            ActionExecuted(some_other_action),
+            SlotSet(some_other_slot, some_other_slot_value),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+    )
+
+    policy = RulePolicy()
+    policy.train(
+        [simple_rule, simple_rule_with_slot_set, simple_story_with_other_slot_set],
+        domain,
+        RegexInterpreter(),
+    )
+
+    conversation_events = [
+        ActionExecuted(ACTION_LISTEN_NAME),
+        UserUttered("haha", {"name": some_intent}),
+    ]
+    prediction = policy.predict_action_probabilities(
+        DialogueStateTracker.from_events(
+            "casd", evts=conversation_events, slots=domain.slots
+        ),
+        domain,
+        RegexInterpreter(),
+    )
+    assert_predicted_action(prediction, domain, some_action)
```

Comment:
`some_slot` should also be reset in the states for ml policy right? Can we check that just as how we check inside `test_hide_rule_turn`

Reply:
updated the test